tools/crm_resource_runtime.c

```diff
@@ -253,97 +248,111 @@ cli_resource_update_attribute(resource_t *rsc, const char *requested_name,
                 return -ENOTUNIQ;
             }
         }
+        resources = g_list_append(resources, rsc);
 
     } else {
-        rsc = find_matching_attr_resource(rsc, requested_name, attr_set,
-                                          attr_id, attr_name, cib, "update");
+        resources = find_matching_attr_resources(rsc, requested_name, attr_set,
+                                                 attr_id, attr_name, cib, "update");
     }
 
-    lookup_id = clone_strip(rsc->id); /* Could be a cloned group! */
-    rc = find_resource_attr(cib, XML_ATTR_ID, lookup_id, attr_set_type, attr_set, attr_id, attr_name,
-                            &local_attr_id);
+    for(GList *gIter = resources; gIter; gIter = gIter->next) {
+        char *lookup_id = NULL;
 
-    if (rc == pcmk_ok) {
-        crm_debug("Found a match for name=%s: id=%s", attr_name, local_attr_id);
-        attr_id = local_attr_id;
+        xmlNode *xml_top = NULL;
+        xmlNode *xml_obj = NULL;
+        local_attr_id = NULL;
+        local_attr_set = NULL;
 
-    } else if (rc != -ENXIO) {
-        free(lookup_id);
-        free(local_attr_id);
-        return rc;
+        rsc = (resource_t*)gIter->data;
+        attr_id = common_attr_id;
 
-    } else {
-        const char *tag = crm_element_name(rsc->xml);
+        lookup_id = clone_strip(rsc->id); /* Could be a cloned group! */
+        rc = find_resource_attr(cib, XML_ATTR_ID, lookup_id, attr_set_type, attr_set, attr_id, attr_name,
+                                &local_attr_id);
 
-        if (attr_set == NULL) {
-            local_attr_set = crm_concat(lookup_id, attr_set_type, '-');
-            attr_set = local_attr_set;
-        }
-        if (attr_id == NULL) {
-            local_attr_id = crm_concat(attr_set, attr_name, '-');
+        if (rc == pcmk_ok) {
+            crm_debug("Found a match for name=%s: id=%s", attr_name, local_attr_id);
             attr_id = local_attr_id;
-        }
 
-        xml_top = create_xml_node(NULL, tag);
-        crm_xml_add(xml_top, XML_ATTR_ID, lookup_id);
+        } else if (rc != -ENXIO) {
+            free(lookup_id);
+            free(local_attr_id);
+            g_list_free(resources);
+            return rc;
 
-        xml_obj = create_xml_node(xml_top, attr_set_type);
-        crm_xml_add(xml_obj, XML_ATTR_ID, attr_set);
-    }
+        } else {
+            const char *tag = crm_element_name(rsc->xml);
 
-    xml_obj = crm_create_nvpair_xml(xml_obj, attr_id, attr_name, attr_value);
-    if (xml_top == NULL) {
-        xml_top = xml_obj;
-    }
+            if (attr_set == NULL) {
+                local_attr_set = crm_concat(lookup_id, attr_set_type, '-');
+                attr_set = local_attr_set;
+            }
+            if (attr_id == NULL) {
+                local_attr_id = crm_concat(attr_set, attr_name, '-');
+                attr_id = local_attr_id;
+            }
```

Comment:
The `local_attr_id` is reset on each iteration to NULL (line 263). 
The line 291 is hit at most once because the `find_matching_attr_resources` function returns only those children that have the requested attribute. For example if the user wants to change the maintenance property of a group of two primitives, where only one has the maintenance attribute set, only that primitive will be added to the list. The group itself will be added unconditionally. Thus the only resource that wouldn't have the requested property would be the group.
I see it works as before with the difference that the children resources that do have the requested property will inherit the new value from their parent.

Reply:
I get it now, thanks